### ISSUE TITLE
TooManyResourcesException thrown regularly

### ISSUE THREAD

**Issue by Akrabor** [EvoSuite/evosuite#7]
I get this exception regularly trying to execute my generated test suites. Regularly means about one or two in 200 test methods which is quite inconvenient when the whole project is much larger than that.
Same setup as in https://github.com/EvoSuite/evosuite/issues/6:

ubuntu 14.04,
java 1.8 32 bit
ant 1.9.3
evosuite 1.0.1

My loop execution limit is at 50000. Disabling it completely results in a timeout... Is there a convenient way to fix/prevent this in general?

**Comment by arcuri82** [comment]
I am not sure if I m fully understanding the problem you are having. TooManyResourcesException is a feature that makes the tests not running for too long, as necessary for unit tests. However, even if such an exception is thrown, the generated tests should catch it and do not fail. Why is it "inconvenient" for you?

**Comment by Akrabor** [comment]
Thanks for your answer. 

Unfortunately the exception does not get caught within the tests, so running them results in errors... As an example:
```
@Test
public void test12()  throws Throwable  {
      StringMap.Entry stringMap_Entry0 = new StringMap.Entry((-2556), "L5_ecdP", (Object) "L5_ecdP", (StringMap.Entry) null);
      stringMap_Entry0.key = "L5_ecdP";
      StringMap.Entry stringMap_Entry1 = new StringMap.Entry((-2556), "L5_ecdP", (Object) stringMap_Entry0, stringMap_Entry0);
      stringMap_Entry1.key = "L5_ecdP";
      stringMap_Entry1.key = "L5_ecdP";
      Object object0 = new Object();
      stringMap_Entry1.value = object0;
      String string0 = stringMap_Entry0.key;
      Object object1 = new Object();
      stringMap_Entry1.value = object1;
      StringMap stringMap0 = new StringMap();
      stringMap0.count = (-2556);
      StringMap stringMap1 = new StringMap(stringMap0);
      StringMap.Entry[] stringMap_EntryArray0 = new StringMap.Entry[4];
      stringMap_EntryArray0[0] = stringMap_Entry1;
      stringMap_EntryArray0[1] = stringMap_Entry1;
      stringMap_EntryArray0[2] = stringMap_Entry0;
      stringMap1.table = stringMap_EntryArray0;
      String string1 = stringMap_Entry1.key;
      stringMap1.put("/H\"io_", stringMap_Entry0);
      stringMap_Entry0.next = stringMap_Entry1;
      StringMap stringMap2 = new StringMap(stringMap1);
  }
```

which means the exception is forwarded to junit. The test of course fails and produces a report entry like:

```
<testcase name="test12" classname="org.apache.batik.css.engine.value.StringMap_ESTest" time="0.077">
    <error message="Loop has been executed more times than the allowed 50000" type="org.evosuite.runtime.TooManyResourcesException">org.evosuite.runtime.TooManyResourcesException: Loop has been executed more times than the allowed 50000
	at org.evosuite.runtime.LoopCounter.checkLoop(LoopCounter.java:102)
	at org.apache.batik.css.engine.value.StringMap.<init>(Unknown Source)
	at org.apache.batik.css.engine.value.StringMap_ESTest.test12(Unknown Source)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
    </error>
</testcase>
```

**Comment by arcuri82** [comment]
ah, yes, now I remember. this is a bug in 1.0.1 that is already fixed in the current 1.0.2-SNAPSHOT. We ll make a new release in the next few weeks but, if you can't wait, you can clone from Git and build the current SNAPSHOT (executable jar is under Master/target folder)

**Comment by Akrabor** [comment]
That sounds very good, I will try it out. Thanks!
